changedetectionio/flask_app.py

```diff
@@ -542,13 +541,15 @@ def ajax_callback_send_notification_test(watch_uuid=None):
         from changedetectionio.apprise_plugin import apprise_custom_api_call_wrapper
         is_global_settings_form = request.args.get('mode', '') == 'global-settings'
         is_group_settings_form = request.args.get('mode', '') == 'group-settings'
-
         # Use an existing random one on the global/main settings form
-        if not watch_uuid and (is_global_settings_form or is_group_settings_form):
+        if not watch_uuid and (is_global_settings_form or is_group_settings_form) \
+                and datastore.data.get('watching'):
+
             logger.debug(f"Send test notification - Choosing random Watch {watch_uuid}")
             watch_uuid = random.choice(list(datastore.data['watching'].keys()))
-
-        watch = datastore.data['watching'].get(watch_uuid)
+            watch = datastore.data['watching'].get(watch_uuid)
+        else:
+            watch = None
```

Comment:
this broke a lot of stuff that was not tested for, for example, sending a test notification when in restock mode